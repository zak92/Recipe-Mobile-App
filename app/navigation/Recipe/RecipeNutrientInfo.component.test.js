import React from 'react';
import { render } from '@testing-library/react-native'; // json rendering
import RecipeNutrientInfo from './RecipeNutrientInfo.component';


describe('RecipeNutrientInfo', () => {
  it("should match snapshot", () => {
    const snapRecipeNutrientInfo= render(<RecipeNutrientInfo />).toJSON(); // render this file to JSON
    expect(snapRecipeNutrientInfo).toMatchSnapshot(); // this snapshot should match the previous snapshot

  });

  it('displays the passed-in calories amount', () => {
    const { queryByText } = render(
      <RecipeNutrientInfo calories="1440" />
    );
    expect(queryByText('1440')).not.toBeNull();
  });

  it('displays the passed-in energy amount', () => {
    const { queryByText } = render(
      <RecipeNutrientInfo energy="1440" />
    );
    expect(queryByText('1440')).not.toBeNull();
  });

  it('displays the passed-in protein amount', () => {
    const { queryByText } = render(
      <RecipeNutrientInfo protein="2550" />
    );
    expect(queryByText('2550')).not.toBeNull();
  });

  it('displays the passed-in carbs amount', () => {
    const { queryByText } = render(
      <RecipeNutrientInfo carbs="5240" />
    );
    expect(queryByText('5240')).not.toBeNull();
  });

  it('displays the passed-in sodium amount', () => {
    const { queryByText } = render(
      <RecipeNutrientInfo sodium="224" />
    );
    expect(queryByText('224')).not.toBeNull();
  });

  it('displays the passed-in cholesterol amount', () => {
    const { queryByText } = render(
      <RecipeNutrientInfo cholesterol="0" />
    );
    expect(queryByText('0')).not.toBeNull();
  });

});