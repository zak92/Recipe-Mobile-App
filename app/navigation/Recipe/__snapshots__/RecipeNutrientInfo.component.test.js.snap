// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeNutrientInfo should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ff6c9b",
        "borderColor": "#ff6c9b",
        "borderWidth": 2,
        "margin": 5,
        "padding": "1%",
        "width": "25%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
      testID="calories"
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 16,
          "paddingTop": "2%",
          "textAlign": "center",
        }
      }
    >
      Calories
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#ff6c9b",
        "borderColor": "#ff6c9b",
        "borderWidth": 2,
        "margin": 5,
        "padding": "1%",
        "width": "25%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
      testID="energy"
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 16,
          "paddingTop": "2%",
          "textAlign": "center",
        }
      }
    >
      Energy(kcal)
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#ff6c9b",
        "borderColor": "#ff6c9b",
        "borderWidth": 2,
        "margin": 5,
        "padding": "1%",
        "width": "25%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
      testID="protein"
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 16,
          "paddingTop": "2%",
          "textAlign": "center",
        }
      }
    >
      Protein(g)
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#ff6c9b",
        "borderColor": "#ff6c9b",
        "borderWidth": 2,
        "margin": 5,
        "padding": "1%",
        "width": "25%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
      testID="carbs"
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 16,
          "paddingTop": "2%",
          "textAlign": "center",
        }
      }
    >
      Carbs(g)
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#ff6c9b",
        "borderColor": "#ff6c9b",
        "borderWidth": 2,
        "margin": 5,
        "padding": "1%",
        "width": "25%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
      testID="sodium"
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 16,
          "paddingTop": "2%",
          "textAlign": "center",
        }
      }
    >
      Sodium(mg)
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#ff6c9b",
        "borderColor": "#ff6c9b",
        "borderWidth": 2,
        "margin": 5,
        "padding": "1%",
        "width": "25%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
      testID="cholesterol"
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 16,
          "paddingTop": "2%",
          "textAlign": "center",
        }
      }
    >
      Cholesterol(mg)
    </Text>
  </View>
</View>
`;
